File: RxTools-library/src/main/java/com/vondear/rxtools/view/dialog/RxDialogShapeLoading.java
package com.vondear.rxtools.view.dialog;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;

import com.vondear.rxtools.R;
import com.vondear.rxtools.view.RxToast;
import com.vondear.rxtools.view.dialog.shapeloadingview.RxShapeLoadingView;

/**
 * @author Vondear
 * @date 2017/3/16
 */

public class RxDialogShapeLoading extends RxDialog {

    private RxShapeLoadingView mLoadingView;
    private View mDialogContentView;

    public RxDialogShapeLoading(Context context, int themeResId) {
        super(context, themeResId);
        initView(context);
    }

    public RxDialogShapeLoading(Context context, boolean cancelable, OnCancelListener cancelListener) {
        super(context, cancelable, cancelListener);
        initView(context);
    }

    public RxDialogShapeLoading(Context context) {
        super(context);
        initView(context);
    }

    public RxDialogShapeLoading(Activity context) {
        super(context);
        initView(context);
    }

    public RxDialogShapeLoading(Context context, float alpha, int gravity) {
        super(context, alpha, gravity);
        initView(context);
    }

    private void initView(Context context) {
        mDialogContentView = LayoutInflater.from(context).inflate(R.layout.dialog_shape_loading_view, null);
        mLoadingView = (RxShapeLoadingView) mDialogContentView.findViewById(R.id.loadView);
        setContentView(mDialogContentView);
    }

    public void cancel(RxCancelType code, String str) {
        cancel();
        switch (code) {
            case normal:
                RxToast.normal(str);
                break;
            case error:
                RxToast.error(str);
                break;
            case success:
                RxToast.success(str);
                break;
            case info:
                RxToast.info(str);
                break;
            default:
                RxToast.normal(str);
                break;
        }
    }

    public void cancel(String str) {
        cancel();
        RxToast.normal(str);
    }

    public void setLoadingText(CharSequence charSequence) {
        mLoadingView.setLoadingText(charSequence);
    }

    public RxShapeLoadingView getLoadingView() {
        return mLoadingView;
    }

    public View getDialogContentView() {
        return mDialogContentView;
    }

    public enum RxCancelType {normal, error, success, info}
}

File: RxUI/src/main/java/com/tamsiree/rxui/view/dialog/RxDialogShapeLoading.kt
package com.tamsiree.rxui.view.dialog

import android.app.Activity
import android.content.Context
import android.content.DialogInterface
import android.view.LayoutInflater
import android.view.View
import com.tamsiree.rxkit.view.RxToast.error
import com.tamsiree.rxkit.view.RxToast.info
import com.tamsiree.rxkit.view.RxToast.normal
import com.tamsiree.rxkit.view.RxToast.success
import com.tamsiree.rxui.R
import com.tamsiree.rxui.view.dialog.shapeloadingview.RxShapeLoadingView

/**
 * @author Tamsiree
 * @date 2017/3/16
 */
class RxDialogShapeLoading : RxDialog {
    lateinit var loadingView: RxShapeLoadingView
        private set
    lateinit var dialogContentView: View
        private set

    constructor(context: Context, themeResId: Int) : super(context, themeResId) {
        initView(context)
    }

    constructor(context: Context, cancelable: Boolean, cancelListener: DialogInterface.OnCancelListener?) : super(context, cancelable, cancelListener) {
        initView(context)
    }

    constructor(context: Context) : super(context) {
        initView(context)
    }

    constructor(context: Activity) : super(context) {
        initView(context)
    }

    constructor(context: Context, alpha: Float, gravity: Int) : super(context, alpha, gravity) {
        initView(context)
    }

    private fun initView(context: Context) {
        dialogContentView = LayoutInflater.from(context).inflate(R.layout.dialog_shape_loading_view, null)
        loadingView = dialogContentView.findViewById(R.id.loadView)
        setContentView(dialogContentView)
    }

    fun cancel(code: RxCancelType?, str: String?) {
        cancel()
        when (code) {
            RxCancelType.normal -> normal(str!!)
            RxCancelType.error -> error(str!!)
            RxCancelType.success -> success(str!!)
            RxCancelType.info -> info(str!!)
            else -> normal(str!!)
        }
    }

    fun cancel(str: String?) {
        cancel()
        normal(str!!)
    }

    fun setLoadingText(charSequence: CharSequence?) {
        loadingView.setLoadingText(charSequence)
    }

    enum class RxCancelType {
        normal, error, success, info
    }
}